### PR TITLE
Close #3186 Use Composer to track az-icons package,  and  add ability for site owner to choose how az-icon library is loaded.

### LINKED ISSUE #3186: Use composer to track the az-icons project instead of managing az-icons artifact in this repo.
At the moment this repository tracks a copy of the az-digital/az-icons package which is not ideal.
Let's change that by using composer to track this repository's contents.

Additionally, I'd like to be able to configure where I am loading the icons from.

### PR DESCRIPTION
**Pull request by trackleft**
## Description
In this PR I am 
1. Adding a new github action that utilizes repository dispatch paradigm to create a new PR upon new release of az-icons. (this matches what we do for [arizona-bootstrap](https://github.com/az-digital/az_quickstart/blob/main/.github/workflows/arizona-bootstrap-release.yml))
2.Removing the copy of the icon font from this repository
3. Adding theme settings and defaults for controlling the output of the library.
4. Update composer.json so that it downloads az-icons

## Related issues
Close #3186

## How to test
Navigate to theme settings for az_barrio
Adjust the settings for az-icons 
image
Save
Navigate to front end, and check that your settings are being respected.

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: themes/custom/az_barrio/az_barrio.libraries.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ az-proxima-nova:
 az-icons:
   css:
     theme:
-      https://cdn.digital.arizona.edu/lib/az-icons/main/az-icons-styles.css: {}
+      /libraries/arizona-bootstrap/css/az-icons-styles.min.css: {}
```
Comment (trackleft) [inline themes/custom/az_barrio/az_barrio.libraries.yml:25]:
```suggestion
      /libraries/az-icons/az-icons-styles.min.css: {}
```

File: composer.json [hunk 2]
```diff
@@ -31,6 +31,7 @@
     "require": {
         "php": ">=7.3",
         "az-digital/arizona-bootstrap": "2.0.13",
+        "az-digital/az-icons": "1.0.0",
```
Comment (trackleft) [inline composer.json:34]:
```suggestion
        "az-digital/az-icons": "1.x-dev",
```

**Comment by trackleft** [comment]
Blocked by https://github.com/az-digital/az-icons/pull/22

File: themes/custom/az_barrio/config/install/az_barrio.settings.yml [hunk 3]
```diff
@@ -86,6 +86,8 @@ az_bootstrap_cdn_version: stable
 az_bootstrap_minified: true
 az_barrio_az_icons: true
 az_barrio_az_icons_source: cdn
+az_barrio_az_icons_minified: true
+az_icons_cdn_version: 1.0.0
```
Comment (trackleft) [inline themes/custom/az_barrio/config/install/az_barrio.settings.yml:90]:
```suggestion
az_icons_cdn_version: stable
```

**Review by trackleft — COMMENTED** [review]
Minor updates required

File: composer.json [hunk 4]
```diff
@@ -31,6 +31,7 @@
     "require": {
         "php": ">=8.1",
         "az-digital/arizona-bootstrap": "2.0.25",
+        "az-digital/az-icons": "1.0.0",
```
Comment (trackleft) [inline composer.json:34]:
```suggestion
        "az-digital/az-icons": "1.0.1",
```

File: themes/custom/az_barrio/includes/common.inc [hunk 5]
```diff
@@ -8,7 +8,7 @@
  */
 
 define('AZ_BOOTSTRAP_STABLE_VERSION', '2.0.25');
-define('AZ_ICONS_STABLE_VERSION', 'main');
+define('AZ_ICONS_STABLE_VERSION', '1.0.0');
```
Comment (trackleft) [inline themes/custom/az_barrio/includes/common.inc:11]:
```suggestion
define('AZ_ICONS_STABLE_VERSION', '1.0.1');
```

**Review by joeparsons — APPROVED** [review]

File: themes/custom/az_barrio/az_barrio.theme [hunk 6]
```diff
@@ -384,14 +384,25 @@ function az_barrio_az_bootstrap_assets_path($type) {
...
-    $az_icons_path = 'https://cdn.digital.arizona.edu/lib/az-icons/' . AZ_ICONS_STABLE_VERSION . '/az-icons-styles.css';
+    $az_icons_path = 'https://cdn.digital.arizona.edu/lib/az-icons/'  . $az_icons_cdn_version;
   }
   else {
-    $az_icons_path = 'libraries/az-icons/az-icons-styles.css';
+    $az_icons_path = base_path() . 'libraries/az-icons';
+  }
+  $az_icons_path .= '/az-icons-styles';
+
+  // AZ Bootstrap minified.
```
Comment (bberndt-uaz) [inline themes/custom/az_barrio/az_barrio.theme:401]:
```suggestion
  // AZ Icons minified.
```

**Review by bberndt-uaz — DISMISSED** [review]

**Review by joeparsons — APPROVED** [review]

File: themes/custom/az_barrio/config/install/az_barrio.settings.yml [hunk 7]
```diff
@@ -86,6 +86,8 @@ az_bootstrap_cdn_version: stable
 az_bootstrap_minified: true
 az_barrio_az_icons: true
 az_barrio_az_icons_source: cdn
+az_barrio_az_icons_minified: true
```
Comment (trackleft) [inline themes/custom/az_barrio/config/install/az_barrio.settings.yml:89]: I don't think this was set when I was reviewing.

File: themes/custom/az_barrio/config/schema/az_barrio.schema.yml [hunk 8]
```diff
@@ -327,10 +327,16 @@ az_barrio.settings:
       label: 'Use Material Design Sharp Icons'
     az_barrio_az_icons:
       type: boolean
-      label: 'Use AZ Icons'
+      label: 'Use Arizona Icons'
     az_barrio_az_icons_source:
       type: string
-      label: 'AZ Icons source'
+      label: 'Arizona Icons source'
+    az_barrio_az_icons_minified:
```
Comment (trackleft) [inline themes/custom/az_barrio/config/schema/az_barrio.schema.yml:334]:
```suggestion
    az_icons_minified:
```

File: themes/custom/az_barrio/config/install/az_barrio.settings.yml (hunk 7 above)
Comment (trackleft) [inline themes/custom/az_barrio/config/install/az_barrio.settings.yml:89]:
```suggestion
az_icons_minifie: true
```

**Comment by trackleft** [comment]
Thanks @tadean , your suggestion to test on an existing site uncovered an issue with setting machine names.  Updated here https://github.com/az-digital/az_quickstart/pull/1465/commits/0b15d76c16509787608506008237db8848ca3eee

I tested on https://icons-azs-eller.pantheonsite.io

**Review by bberndt-uaz — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**